src/main/java/com/rultor/profiles/Profiles.java

```diff
@@ -73,18 +73,93 @@ public Profile fetch(final Talk talk) throws IOException {
             if (xml.nodes("/talk/wire").isEmpty()) {
                 profile = Profile.EMPTY;
             } else {
-                profile = Profiles.fetch(xml);
+                profile = this.fetch(xml);
             }
         }
         return profile;
     }
 
     /**
+     * Merge profile from master and fork. Merged profile must be profile
+     * from fork, but all lists for commanders and architects must be taken
+     * from master branch.
+     * @param master Profile from master branch
+     * @param fork Fork that will be merged
+     * @param branch Fork branch that will be merged
+     * @return Merged profile
+     * @todo #1039:30min Modify method implementation: merged profile is a
+     *  `.rultor.yml` form result of real `git merge` of master branch and
+     *  fork branch. We need real `git merge` result here. See some details
+     *  here: https://github.com/yegor256/rultor/pull/1064/files#r56796959
+     */
+    public Profile merged(
+        final Profile master,
+        final String fork,
+        final String branch
+    ) {
+        return master;
+    }
+
+    /**
+     * Validate merged profile: merged profile must have no changes in architect
+     * and commander sections (full match with master profile).
+     * @param master Profile from master branch
+     * @param merged Result of merging master and fork profiles
+     * @return Validated profile
+     * @throws IOException If fails
+     * @checkstyle MultipleStringLiteralsCheck (10 lines)
+     */
+    public Profile validated(final Profile master, final Profile merged)
+        throws IOException {
+        final String[][] security = {
+            {"architect"},
+            {"merge", "commanders"},
+            {"deploy", "commanders"},
+            {"release", "commanders"},
+        };
+        for (final String[] section : security) {
+            if (!Arrays.equals(
+                section(master, section),
+                section(merged, section)
+            )) {
+                throw new Profile.ConfigException(
+                    String.format(
+                        "You cannot change `%s` section for security reasons",
+                        Arrays.toString(section)
+                    )
+                );
+            }
+        }
+        return merged;
+    }
+
+    /**
+     * Get section content.
+     * @param profile Profile
+     * @param section Section of profile
+     * @return Content of section as array of strings
+     * @throws IOException If fails
+     */
+    private String[] section(final Profile profile, final String... section)
```

Comment:
@mkordas No. Because String array is a section, described as sequence of nested entries.